assets/src/components/d3/createProgressBar.js

```diff
@@ -72,46 +72,60 @@ function createProgressBar ({ data, width, height, el, tip }) {
 
   const currentIndex = data.filter(x => x.graded).length
   const currentLine = svg.append('g')
-
-  currentLine
-    .append('line')
-    .attr('x1', x(calculatePercentSoFar(currentIndex)))
-    .attr('x2', x(calculatePercentSoFar(currentIndex)))
-    .attr('y1', 0)
-    .attr('y2', aHeight - margin.bottom)
-    .attr('stroke', 'darkorange')
-    .attr('stroke-width', '1')
-
-  currentLine.append('text')
-    .attr('text-anchor', 'middle')
-    .attr('x', x(calculatePercentSoFar(currentIndex)) - 26)
-    .attr('y', margin.top)
-    .attr('dy', -7)
-    .attr('fill', 'darkorange')
-    .attr('font-size', '12px')
-    .attr('font-weight', 'bold')
-    .text('Current')
+   // only show currentline when assignments are graded
+  if(currentIndex != 0) {
+    currentLine
+      .append('line')
+      .attr('x1', x(calculatePercentSoFar(currentIndex)))
+      .attr('x2', x(calculatePercentSoFar(currentIndex)))
+      .attr('y1', 0)
+      .attr('y2', aHeight - margin.bottom)
+      .attr('stroke', 'darkorange')
+      .attr('stroke-width', '1')
+
+    currentLine.append('text')
+      .attr('text-anchor', 'middle')
+      .attr('x', x(calculatePercentSoFar(currentIndex)) - 26)
+      .attr('y', margin.top)
+      .attr('dy', -7)
+      .attr('fill', 'darkorange')
+      .attr('font-size', '12px')
+      .attr('font-weight', 'bold')
+      .text('Current')
+  }
 
   const maxLine = svg.append('g')
 
-  maxLine
-    .append('line')
-    .attr('x1', x(calculatePercentSoFar(data.length)))
-    .attr('x2', x(calculatePercentSoFar(data.length)))
-    .attr('y1', 0)
-    .attr('y2', aHeight - margin.bottom)
-    .attr('stroke', 'green')
-    .attr('stroke-width', '1')
-
-  maxLine.append('text')
-    .attr('text-anchor', 'middle')
-    .attr('x', x(calculatePercentSoFar(data.length)) - 42)
-    .attr('y', margin.top)
-    .attr('dy', -7)
-    .attr('fill', 'green')
-    .attr('font-size', '12px')
-    .attr('font-weight', 'bold')
-    .text('Max Possible')
+  // login to show the maxLine based on the graded and ungraded data
+  const showMaxLine = (data) => {
+    // filtering the graded and ungraded values, list would be either 2 or 1
+    const gradedList = [...new Set(data.map(item => item.graded))]
+    if(gradedList.length >1){
+      return false
+    }
+    return gradedList[0]
+  }
+
+  if(!showMaxLine(data)) {
```

Comment:
Could you explain what `showMaxLine` does?

I'm a bit confused because it looks like `maxLine` is appended only if `showMaxLine(data)` _isn't_ true. Shouldn't it be appended if `showMaxLine(data)` _is_ true?

Reply:
MaxLine should be shown
1. When you have a mix of graded and ungraded assignment, this is usually in the mids of the term
2. When No Graded assignments at all, this is the case for admin/teacher whose assignment won't be graded

Similarly, CurrentLine should be shown
1. when data has a mix of graded and ungraded assignment
2. When all assignments are graded.

Reply:
OK, I changed the logic to make it less confusing

Reply:
Perfect, makes sense!